security/fma-kona-asterisc.md

```diff
@@ -0,0 +1,213 @@
+  When games are detected on-chain, the `op-challenger` will create a "game player" for this game type.
+  Players are run in individual threads so a breaking `kona` + `asterisc` game player in the challenger will not cause a liveness issue for the `op-challenger` to play other game types.
+  Another key mitigation is the VM runner. This is an offline simulation that runs asterisc + kona with the op-challenger. It's been running since around August 2024.
+
+- **Detection:**
+  The Dispute Monitor ([op-dispute-mon] service provides an isolated monitoring service that will detect if a game is not being played by the honest `op-challenger`.
+  It will alert in this case.
+
+- **Recovery Path(s):**
+  If the `op-challenger` fails to play `kona` + `asterisc` fault dispute games, a fix will need to be made to the `op-challenger`.
+  It does not require governance, but would need a new release of the `op-challenger` to be published and rolled out to infrastructure.
+
+[op-dispute-mon]: https://github.com/ethereum-optimism/optimism/tree/develop/op-dispute-mon
+[Game Type 3]: https://github.com/ethereum-optimism/optimism/blob/develop/op-challenger/game/fault/types/types.go#L30
+
+
+### Divergence between RISCV.sol and Asterisc
+
+- **Description:**
+  Asterisc is a "Fault Proof VM".
+  It is an onchain RISCV instruction emulator with the on-chain component of the kona + asterisc fault dispute game being the `RISCV.sol` contract.
+  `RISCV.sol` runs a single RISCV instruction on chain with merkleized VM memory.
+  Effectively, the behavior between `RISCV.sol` and asterisc cannot diverge.
+
+- **Risk Assessment:**
+  Low likelihood, but high impact.
+
+- **Mitigation:**
+  Heavily test `RISCV.sol`. Audit `RISCV.sol`.
+  Asterisc already runs through action tests frequently in CI validiting that it produces the expected output.
+
+- **Detection:**
+  Since `op-dispute-mon` uses the same asterisc backend that the `op-challenger` does, there may be divergent behaviour between asterisc and `RISCV.sol`.
+  In the worst case, if a silent issue is exploited in `RISCV.sol`, a fault dispute game may resolve incorrectly.
+  But since the kona + asterisc fault dispute game lives outside the hotpath, this game result is not used anywhere by the portal and can be discarded.
+
+- **Recovery Path(s):**
+  If a divergence between `RISCV.sol` and asterisc is found, either or both would need to be updated, and re-released.
+  Since the kona + asterisc Fault Dispute Game would need to use the updated `RISCV.sol`, if it was fixed, an upgrade would need to be performed.
+  If the kona + asterisc fault dispute game sits in the hotpath, the security council would need to fallback to the op-program + cannon fault dispute
+  game, and only switch back once the fix has been released and deployed.
+
+
+### Generic items we need to take into account:
```

Comment:
From @clabby's suggestion https://github.com/ethereum-optimism/design-docs/pull/140#discussion_r1817421106, is there somewhere in here that discusses the FMA-specific mitigations for the generic items listed here?

Reply:
Done